### PR TITLE
Deploy docs

### PR DESCRIPTION
**Pull request by lilyminium**
Add CI for doc deployment

Sorry, I will comment later, there is a turducken

### CODE REVIEW

**Comment by PicoCentauri** [comment]
Thanks for the work @lilyminium. 

For the build job to be passed add something like 

```rst
new_version
------------------------------------------

 *Deploy docs (#75)

```
to `docs/CHANGELOG.rst`. An example is also given in [prior PRs](https://github.com/MDAnalysis/mdacli/pull/75/files#diff-a3eb23aaf44c5fba1fabdb90f9d57bc4d0ca6ccfecc4374b45835880a3d56934).

File: docs/CHANGELOG.rst [hunk 1]
```diff
@@ -2,6 +2,11 @@
 Changelog
 =========
 
+v0.1.5 (2021-11-25)
```
Comment (PicoCentauri) [inline docs/CHANGELOG.rst:5]: this should be `new_version`. Nobody knows if the next version is really 0.1.5. Maybe there is release before that one. ;-)

```suggestion
new_version
```

**Review by PicoCentauri — CHANGES_REQUESTED** [review]

File: docs/CHANGELOG.rst (hunk 1 above)
Comment (lilyminium) [reply docs/CHANGELOG.rst]: Ah, that makes sense, thanks!

**Comment by lilyminium** [comment]
If I'm understanding the library correctly, at the moment, every new push to `main` will register as the new `stable` and `dev` versions, because the `mdacli.__version__` won't have any kind of dev label (e.g. MDA itself uses 2.0.0-dev to signify it's not the real 2.0.0 release). I *could* fix this by grepping CHANGELOG.rst instead of `__init__.py`, but I wondered if you guys might want to change how the `__version__` currently works, because IMO it's currently inaccurate if I'm working off a development version.

I added some comments to the script and will do a fuller rundown on the wiki by tomorrow / the weekend.

**Comment by lilyminium** [comment]
Just putting this PR on draft status; I'm remembering various other extensions that are needed for installation as I write the wiki.

**Comment by lilyminium** [comment]
https://mdacli.minium.com.au/ is an example of the built docs, from the PR https://github.com/lilyminium/mdacli/pull/1 . You can see the updated gh-pages branch here: https://github.com/lilyminium/mdacli/tree/gh-pages

A wiki page explaining what's happening is available here: https://github.com/MDAnalysis/mdanalysis/wiki/Documentation-deployment

**Comment by orbeckst** [comment]
@PicoCentauri do you want to finish this PR before letting the blog post go live https://github.com/MDAnalysis/MDAnalysis.github.io/pull/207 ?

**Review by PicoCentauri — APPROVED** [review]

**Pull request merged**